### PR TITLE
DOC: fix RT03, ES01 for pandas.core.groupby.DataFrameGroupBy.agg and pandas.core.groupby.DataFrameGroupBy.aggregate

### PR DESCRIPTION
**Pull request by tuhinsharma121**
fixes

```
pandas.core.groupby.DataFrameGroupBy.agg RT03, ES01
pandas.core.groupby.DataFrameGroupBy.aggregate RT03, ES01
```

### CODE REVIEW

File: pandas/core/groupby/generic.py [hunk 1]
```diff
@@ -67,7 +67,6 @@
 from pandas.core.groupby.groupby import (
     GroupBy,
     GroupByPlot,
-    _agg_template_frame,
```
Comment (rhshadrach) [inline pandas/core/groupby/generic.py:70]: Can `_agg_template_frame` be removed from `groupby.py`?

**Review by rhshadrach — COMMENTED** [review]
Thanks for the PR!

File: pandas/core/groupby/generic.py (hunk 1 above)
Comment (tuhinsharma121) [reply pandas/core/groupby/generic.py:70]: yeah. Its being used here - https://github.com/pandas-dev/pandas/pull/59869/files#diff-f1ec980d06b0b54c8263f663f767636c3f5921f04f8a7ee91dfc71a2100b05cdL327-L328

Comment (rhshadrach) [reply pandas/core/groupby/generic.py:70]: Isn't that `_agg_template_series` and not `_agg_template_frame`?

Comment (tuhinsharma121) [reply pandas/core/groupby/generic.py:70]: You are right. I got confused. `_agg_template_frame ` was only referenced at line - 1518 and we can remove it -  https://github.com/pandas-dev/pandas/pull/59869/files#diff-f1ec980d06b0b54c8263f663f767636c3f5921f04f8a7ee91dfc71a2100b05cdL1518

Comment (tuhinsharma121) [reply pandas/core/groupby/generic.py:70]: I think its good to merge. Let me know if I need to change anything?

Comment (mroeschke) [reply pandas/core/groupby/generic.py:70]: Looks like `_agg_template_frame` can still be removed from `groupby.py` and `validate_unwanted_patterns.py`

**Comment by rhshadrach** [comment]
This is looking good to me, but we'll need to wait until the CI gets fixed in order to ensure the docs build correctly.

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @tuhinsharma121

**Pull request merged**